### PR TITLE
Patch for Plugin drivers in docker info 

### PR DESCRIPTION
**Pull request by kunalkushwaha**
Supported added for retrieving Plugin list for Network and Volume.
Also, plugin information in docker info output. https://github.com/docker/docker/issues/16205

Signed-off-by: Kunal Kushwaha kushwaha_kunal_v7@lab.ntt.co.jp

### CODE REVIEW

**Comment by calavera** [comment]
I'm not sure if we want to do this yet, it looks like a good idea. But if we do it, the api response should be structured, not a string with all the names.

**Comment by kunalkushwaha** [comment]
Since, most people were positive in discussion for #16205 Proposal, so created patch for same.

For structured API output, what all information will be useful to be included? Driver themselves doesn't have many properties, volumes have.

**Comment by calavera** [comment]
we can start with simple arrays with the names, pretty much like you did with a small change, something like this:

```json
{
  "Plugins": {
     "Volume": ["local", "foo", "bar"...],
     "Network": ["local", "overlay", ....]
  }
```

Then you can take that in the client and format it.

File: volume/drivers/extpoint.go [hunk 1]
```diff
@@ -106,3 +106,13 @@ func GetDriver(name string) (volume.Driver, error) {
 	}
 	return Lookup(name)
 }
+
+// GetDriverList returns list of volume drivers registered.
+// If no driver is registered, empty string list will be returned.
+func GetDriverList() []string {
+	var driverList []string
+	for driverName, _ := range drivers.extensions {
```
Comment (wenchma) [inline volume/drivers/extpoint.go]: @kunalkushwaha `golint` errors:  no need to use underscore `_` in the variable

Comment (wenchma) [inline volume/drivers/extpoint.go]: @kunalkushwaha  golint error: no need to use underscore `_` in the variable

File: daemon/info.go [hunk 2]
```diff
@@ -121,3 +123,16 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
...
 	return v, nil
 }
+
+func (daemon *Daemon) showPluginsInfo() types.PluginsInfo {
+	var pluginsInfo types.PluginsInfo
+
+	pluginsInfo.Volume = volumedrivers.GetDriverList()
+
+	networkDriverList := daemon.GetNetworkDriverList()
+	for nd, _ := range networkDriverList {
```
Comment (wenchma) [inline daemon/info.go]: golint error: no need to use underscore `_` in the variable

**Comment by LK4D4** [comment]
@kunalkushwaha Are you still interested in this? Seems like you need to fix gofmt errors.

**Comment by kunalkushwaha** [comment]
Sorry for delay. Made changes to fix golint issues.

**Comment by calavera** [comment]
I like it, thank you! :+1: 

I think we should add some tests for the output, what do you think?

**Comment by kunalkushwaha** [comment]
Thanks, will add the test cases

**Comment by kunalkushwaha** [comment]
Have added code for test in existing function to test ``docker info``.

Have made changes in documents according to new output of ``docker info``.

**Comment by calavera** [comment]
fair enough. LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**